test/cluster/test_tablets_migration.py

```
@@ -767,3 +792,16 @@ async def test_restart_in_cleanup_stage_after_cleanup(manager: ScyllaClusterMana
+        # The tablet's storage group must have been removed from the leaving replica by
+        # now (either at the normal cleanup stage, or -- before the fix for the race this
+        # test targets -- possibly via storage_group_manager::remove_storage_group()
+        # bypassing table::cleanup_tablet()'s own invalidation, if cleanup was still in
+        # flight around the restart). Either way, the row-cache range must end up
+        # invalidated: the shard's cached-partition count must have dropped from what the
+        # warm-up produced (other tablets remaining on the shard keep their own cached
+        # data, so this isn't expected to reach zero).
+        partitions_after_migration = await get_cache_partitions(manager, src_server, src_shard)
+        assert partitions_after_migration < partitions_after_warmup, \
+            f"expected cache partitions on shard {src_shard} of the leaving replica to " \
+            f"drop after tablet migration+cleanup, but {partitions_after_warmup} -> {partitions_after_migration}"
```

Ruff (0.16.3)

[warning] Implicitly concatenated string literals over multiple lines

(ISC002)